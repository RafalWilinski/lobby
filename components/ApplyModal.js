import React from "react";
import { Modal, Button, Input, Alert } from "antd";

const { TextArea } = Input;

class ApplyModal extends React.Component {
  state = {
    desc: ""
  };

  handleOk = () => {
    this.props.apply(this.props.role.id, this.state.desc);
  };

  handleCancel = () => {
    this.props.hideModal();
  };

  render() {
    return (
      <div>
        <Modal
          title={`Aplikacja na stanowisko ${this.props.role &&
            this.props.role.name}`}
          visible={this.props.visible}
          onOk={this.handleOk}
          confirmLoading={this.props.isLoading}
          onCancel={this.handleCancel}
          okText="Aplikuj!"
          cancelText="Anuluj"
        >
          <TextArea
            onChange={e => this.setState({ desc: e.target.value })}
            rows={4}
            ref={r => (this.desc = r)}
            placeholder="Powiedz dlaczego nadajesz się na to stanowisko i dlaczego interesuje Cię ten temat"
          />
          {this.props.error && (
            <Alert message={this.props.error.message} type="error" />
          )}
        </Modal>
      </div>
    );
  }
}

export default ApplyModal;
