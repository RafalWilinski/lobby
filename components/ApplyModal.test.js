import React from "react";
import renderer from "react-test-renderer";
import ApplyModal from "./ApplyModal";

describe("ApplyModal", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ApplyModal />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
