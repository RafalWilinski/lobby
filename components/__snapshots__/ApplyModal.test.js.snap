// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplyModal renders correctly 1`] = `<div />`;
